ramiro_serra/Debug and profiling notes.md

```diff
@@ -0,0 +1,113 @@
+# Debug and profiling notes
+The whole document is based on [this note](http://wiki.ros.org/roslaunch/Tutorials/Roslaunch%20Nodes%20in%20Valgrind%20or%20GDB), so you can check it first.
+To be able to follow the examples, install valgrind and callgrind.
+
+## Using Valgrind for profiling
+
+[This other article](http://wiki.ros.org/roslaunch/Tutorials/Profiling%20roslaunch%20nodes) talks about profiling.
+We'll use the [create_autonomy](https://github.com/RoboticaUtnFrba/create_autonomy) repo, in order to make some examples.
+
+First, we open the docker launchfile, [see here](https://github.com/RoboticaUtnFrba/create_autonomy/wiki/Setup-Docker), then we stall `valgrind` and `kcachegrind`, we will make use of that later. `sudo apt-get install valgrind callgrind -y`
+
+Then, we compile our project, `catkin_make DCMAKE_BUILD_TYPE=Debug`.
+Then, we source devel/setup.bash `source devel/setup.bash`
+Now, we will decide which node to profile, in our case `send_goal`, a node located in `ca_move_base` pkg.
+In order to profile that node, we have to include a `launch-prefix` tag, into the desired `node` tag into the `launch file`.
+So, to locate the launchfile `roscd ca_move_base/launch`.
+We can edit the `send_goal.launch` file with whatever editor we feel like.
+
+This is how my `send_goal.launch` looked like after the changes I made.
+    
+    <launch>
+        <arg name="x" default="2.0"/>
+        <arg name="y" default="2.0"/>
+        <arg name="Y" default="1.0"/>
+        <arg name="ns" default="create1"/>
+        <node pkg="ca_move_base" type="send_goal" name="navigation_goals"
+                args="$(arg x) $(arg y) $(arg Y)" ns="$(arg ns)" launch-prefix="valgrind --tool=callgrind --callgrind-out-file='callgrind.sendgoal.%p'"/>
+    </launch>
+
+Note i added some default values to the required arguments, only to make the usage of the launchfile more straight-forward.
+We can launch now the file with `roslaunch ca_move_base send_goal.launch`, we can let it run a little and then `C-c` to stop it.
+Valgrind will automatically create a log located in `~/.ros/`, let's check it:
+    
+    cd ~/.ros/
+    ls
+
+Then we can open it with `kcachegrind`, that we installed earlier
+    
+    kcachegrind callgrind.sendgoal.10303
+*The appended number may very with the PID associated.*
+
+![kcachegrind](media/kcachegrind.png)
+
+Now a window shows up, with a lot of info of the program, if you want to profundize more on the topic, you can see [here](http://valgrind.org/docs/), and learn how to interpret the data.

```

Comment:
I guess that "profundize" doesn't exist.

Reply:
Changed profundize to `deepen`